src/streaming-client.ts

```diff
@@ -158,6 +163,7 @@ export class RevAiStreamingClient extends EventEmitter {
 
     private closeStreams(): void {
         this.requests.end();
+        this.streamsClosed = true;
```

Comment:
Moved this line from where is was in the suggestion left by Descript. 

There is a still a race here where a message can come in just before this line but not get written to the duplex until after the null is pushed but I think by changing the order of these lines we reduce the likelyhood of this race